src/handlers/keyboard_handler.rs

```diff
@@ -0,0 +1,460 @@
+        Keysym::Num_Lock => Key::NumLock,
+
+        // Numpad Specific Keys (assuming key_char didn't handle them, e.g., if NumLock is off or they are distinct)
+        // Keysym::KP_0 => Key::Numpad0,
+        // Keysym::KP_1 => Key::Numpad1,
+        // Keysym::KP_2 => Key::Numpad2,
+        // Keysym::KP_3 => Key::Numpad3,
+        // Keysym::KP_4 => Key::Numpad4,
+        // Keysym::KP_5 => Key::Numpad5,
+        // Keysym::KP_6 => Key::Numpad6,
+        // Keysym::KP_7 => Key::Numpad7,
+        // Keysym::KP_8 => Key::Numpad8,
+        // Keysym::KP_9 => Key::Numpad9,
+        // Keysym::KP_Add => Key::NumpadAdd,
+        // Keysym::KP_Subtract => Key::NumpadSubtract,
+        // Keysym::KP_Multiply => Key::NumpadMultiply,
+        // Keysym::KP_Divide => Key::NumpadDivide,
+        // Keysym::KP_Decimal => Key::NumpadDecimal,
+        // Keysym::KP_Separator => Key::NumpadComma, // Bevy uses NumpadComma for separator
+        // Keysym::KP_Equal => Key::NumpadEqual,
+        // KP_Enter is handled above with Return
+
+        // Keys that *might* have produced a character but key_char() returned None.
+        // This could happen for dead keys, IME input, or if key_char() implementation is limited.
+        // We need a fallback. Panicking with todo!() highlights these cases during development.
+        // A production system might need more robust handling (e.g., logging, ignoring, specific Key variant if available).
+        Keysym::A
+        | Keysym::B
+        | Keysym::C
+        | Keysym::D
+        | Keysym::E
+        | Keysym::F
+        | Keysym::G
+        | Keysym::H
+        | Keysym::I
+        | Keysym::J
+        | Keysym::K
+        | Keysym::L
+        | Keysym::M
+        | Keysym::N
+        | Keysym::O
+        | Keysym::P
+        | Keysym::Q
+        | Keysym::R
+        | Keysym::S
+        | Keysym::T
+        | Keysym::U
+        | Keysym::V
+        | Keysym::W
+        | Keysym::X
+        | Keysym::Y
+        | Keysym::Z
+        | Keysym::_0
+        | Keysym::_1
+        | Keysym::_2
+        | Keysym::_3
+        | Keysym::_4
+        | Keysym::_5
+        | Keysym::_6
+        | Keysym::_7
+        | Keysym::_8
+        | Keysym::_9
+        | Keysym::grave
+        | Keysym::minus
+        | Keysym::equal
+        | Keysym::bracketleft
+        | Keysym::bracketright
+        | Keysym::backslash
+        | Keysym::semicolon
+        | Keysym::apostrophe
+        | Keysym::comma
+        | Keysym::period
+        | Keysym::slash
+        | Keysym::space => {
+            warn!(
+                "Keysym {:?} did not produce a character via key_char(). This might be unexpected.",
+                keysym
+            );
+            // Bevy's Key enum doesn't have a generic 'Unidentified' or 'Unknown' variant.
+            // Panicking via todo!() helps catch these during development.
+            // You might need custom logic or decide to ignore these cases in production.
+            todo!(
```

Comment:
What are the chances of this happening or in what scenario could it happen, lets discuss this more?